h5p-server/express.js

```diff
@@ -0,0 +1,279 @@
+    );
+
+    // cron job runs once daily at 3:00am to delete temporary files that have expired.
+    // expired time being 24 hours after initial upload
+    cron.schedule("0 3 * * *", () => {
+      h5pEditor.temporaryFileManager.cleanUp();
+    });
+
+    resolve();
+  };
+  return new Promise(setupServerPromise);
+};
+
+new Promise(setupConfig)
+  .then(setupPlayerAndEditor)
+  .then(setupServer)
+  .then(page => {
+    app.get("/status", (req, res) => {
+      res.status(200).end();
+    });
+
+    // only expose /admin route if in development
+    if (process.env.ENVIRONMENT != "prod") {
+      app.get("/admin", (req, res) => {
+        h5pEditor.setRenderer(adminRenderer);
+        h5pEditor
+          .render(undefined, "en", req.user)
+          .then(page => {
+            res.send(page);
+            res.status(200).end();
+          })
+          .catch(error => {
+            console.error("Error GET /admin:");
+            console.error(error);
+            res.status(500).end();
+          });
+      });
+    }
+
+    // create new h5p content of a given type
+    app.get("/new/:type", (req, res) => {
+      h5pEditor.setRenderer(editorRenderer);
+
+      h5pEditor
+        .render(undefined, "en", req.user)
+        .then(page => {
+          res.send(page);
+          res.status(200).end();
+        })
+        .catch(error => {
+          console.error("Error GET /new/:type :");
+          console.error(error);
+          res.status(500).end();
+        });
+    });
+
+    app.get("/edit/:type", (req, res) => {
+      h5pEditor.setRenderer(editorRenderer);
+
+      h5pEditor
+        .render(undefined, "en", req.user)
+        .then(page => {
+          res.send(page);
+          res.status(200).end();
+        })
+        .catch(error => {
+          console.error("Error GET /edit/:type :");
+          console.error(error);
+          res.status(500).end();
+        });
+    });
+
+    app.get(`${h5pEditor.config.playUrl}/:type`, (req, res) => {
+      // TODO provide h5pPlayer with content id depending on h5P | materia toggle? Is that needed?
+      // otherwise, we're looking for req.params.contentId
+      if (req.params.type == "undefined") {
+        // TODO: probly need to display a 404 for this
+        console.error("Error GET /play/:type : type undefined");
+        res.status(404).end();
+      }
+
+      // type directs to render configs located at h5p/content/:type
+      h5pPlayer
+        .render(req.params.type, "en", req.user)
+        .then(h5pPage => {
+          res.send(h5pPage);
+          res.status(200).end();
+        })
+        .catch(error => {
+          console.error("Error GET /play/:type :");
+          console.error(error);
+          res.status(500).end();
+        });
+    });
+
+    // return a new h5p widget
+    // used for materia to check for specific library
+    // should be deprecated as there is no need to save this information on the server
+    app.post("/new/:type", (req, res) => {
```

Comment:
Is this section actually used? My understanding is we're not actually saving H5P content on the server - which I'm pretty sure was the original intent of the `/new/` post request. I'm not sure if Materia actually performs this request as the comment suggests.